### PR TITLE
Simulated mouse movement

### PR DESCRIPTION
**Pull request by jscarty**
Sometimes you need to use the mouse for drag and drop movements to make sure events are triggered.

### CODE REVIEW

File: lib/WWW/WebKit2/MouseInput.pm [hunk 1]
```diff
@@ -325,4 +325,59 @@ sub native_drag_and_drop_to_object {
...
+This method is similar to native_drag_and_drop_to_object but will do the action using X11libs XTest input simulation methods
+
+=cut
+
+
+sub mouse_input_drag_and_drop_to_object {
+    my ($self, $source_locator, $target_locator, $options) =@_;
+
+    croak "did not find element $source_locator" unless $self->resolve_locator($source_locator)->get_length;
+    croak "did not find element $target_locator" unless $self->resolve_locator($target_locator)->get_length;
```
Comment (pvoglhofer) [inline lib/WWW/WebKit2/MouseInput.pm:340]: duplicate line

File: lib/WWW/WebKit2/MouseInput.pm [hunk 2]
```diff
@@ -325,4 +325,59 @@ sub native_drag_and_drop_to_object {
...
+    # otherwise a drop may not work
+    $self->move_mouse_abs($x, $y);
+    $self->pause($step_delay);
+
+    $self->release_mouse_button(1);
+    $self->pause($step_delay);
+    $self->move_mouse_abs($x, $y);
+    $self->pause($step_delay);
+    $self->pause(300);
+    $self->is_loading;
```
Comment (pvoglhofer) [inline lib/WWW/WebKit2/MouseInput.pm]: the $self->is_loading appears to not be doing anything

**Review by pvoglhofer — CHANGES_REQUESTED** [review]

File: lib/WWW/WebKit2/MouseInput.pm (hunk 1 above)
Comment (jscarty) [reply lib/WWW/WebKit2/MouseInput.pm:340]: They're looking for different locators

Comment (pvoglhofer) [reply lib/WWW/WebKit2/MouseInput.pm:340]: 🙈

**Review by pvoglhofer — APPROVED** [review]

**Pull request merged**